back/boxtribute_server/business_logic/warehouse/box/crud.py

```diff
@@ -131,6 +183,47 @@ def update_box(
     Insert timestamp for modification and return the box.
     """
     box = Box.get(Box.label_identifier == label_identifier)
+    box_contains_measure_product = box.size_id is None
+
```

Comment:
Please simplify by not distinguish between productid is None and not None. That actually means that we enforce that a productId is always passed either with a sizeId or a measureValue+DisplayUnit.

` new_product = Product.get_by_id(product_id) or Product.get_by_id(box.product_id)`

Reply:
The flow should be like this:
```
if product_id or size_id or display_unit_id or measureValue is not None:
    new_product = Product.get_by_id(product_id) or Product.get_by_id(box.product_id)
    is_measure_product = ...

    if is_measure_product:
         line 190 - 202
    else:
         line 204 - 207
         if size_id is None: throw error that size_id is required for size Product.
```